templates/chrony.conf.epp

```diff
@@ -0,0 +1,127 @@
+# NTP servers
+<% $chrony::servers.each |$server| { -%>
+server <%= $server.flatten.join(' ') %>
+<% } -%>
+<% $chrony::pools.each |$pool| { -%>
+pool <%= $pool.flatten.join(' ') %>
+<% } -%>
+<% $chrony::peers.each |$peer| { -%>
+peer <%= $peer.flatten.join(' ') %>
+<% } -%>
+
+<% if $chrony::stratumweight { -%>
+
+# How much distance should be added per stratum to the synchronisation distance when
+# chronyd selects the synchronisation source from available sources.
+stratumweight <%= $chrony::stratumweight %>
+<% } -%>
+
+# Record the rate at which the system clock gains/losses time.
+driftfile /var/lib/chrony/drift
+
+# Enable kernel RTC synchronization.
+rtcsync
+<% if $chrony::makestep_seconds and $chrony::makestep_updates { -%>
+
+# In first <%= $chrony::makestep_updates %> updates step the system clock instead of slew
+# if the adjustment is larger than <%= $chrony::makestep_seconds %> seconds.
+makestep <%= $chrony::makestep_seconds %> <%= $chrony::makestep_updates %>
+<% } -%>
+<% if ! $chrony::queryhosts.empty { -%>
+
+# Allow client access.
+<%   $chrony::queryhosts.each |$allowed| { -%>
+allow <%= $allowed %>
+<%   } -%>
+<% } -%>
+<% if $chrony::cmdport { -%>
+cmdport <%= $chrony::cmdport %>
+<% } -%>
+<% if ! $chrony::bindcmdaddress.empty { -%>
+
+<%  $chrony::bindcmdaddress.each |$addr| { -%>
+bindcmdaddress <%= $addr %>
+<%   } -%>
+<% } -%>
+<% if ! $chrony::cmdacl.empty { -%>
```

Comment:
Technically this isn't needed. It's guaranteed to be an array and you can iterate 0 times over an empty array. In other cases it makes sense for whitespace control but that doesn't happen here.

Reply:
You are right again. And you also guessed correctly that my reason for several of the if-statements is to limit empty blank lines in the file.